common/testing.cpp

```diff
@@ -38,6 +42,46 @@ std::string TemporaryDirectory::Path() {
 	return path_;
 }
 
+::testing::AssertionResult FileContains(const string &filename, const string &expected_content) {
+	ifstream is {filename};
+	ostringstream contents_s;
+	contents_s << is.rdbuf();
+	string contents {contents_s.str()};
+	if (contents == expected_content) {
+		return ::testing::AssertionSuccess();
+	}
+	return ::testing::AssertionFailure()
+		   << "Expected: '" << expected_content << "' Got: '" << contents << "'";
+}
+
+
+::testing::AssertionResult FileJsonEquals(const string &filename, const string &expected_content) {
+	ifstream is {filename};
+	json::Json contents = json::Load(is).value();
+	json::Json expected_contents = json::Load(expected_content).value();
+	if (contents.Dump() == expected_contents.Dump()) {
+		return ::testing::AssertionSuccess();
+	}
+	return ::testing::AssertionFailure()
+		   << "Expected: '" << contents.Dump() << "' Got: '" << expected_contents.Dump() << "'";
+}
+
+::testing::AssertionResult FilesEqual(const string &filename1, const string &filename2) {
+	processes::Process proc({"diff", "-u", filename1, filename2});
```

Comment:
For such extremely common options? `diff` doesn't print patches correctly without `-u/--unified`, it's practically part of the command name, and `git diff` doesn't even support non-unified mode. I think I vote against it in this case, since you have to use it in every instance, so it's better to know what it means. 🙂

I wouldn't use `-y` because it's harder to catch small differences when reading side by side, and I also suspect that Gitlab's log output is not wide enough to render them in a readable way.